tests/test_deploy.py

```diff
@@ -0,0 +1,37 @@
+import json
+
+
+def test_deploy_full_deployment(exec_context, action_docker_image_id):
```

Comment:
It looks like there's a lot of conditional branching happening within deploy.sh - maybe something that's already on your todo list, but can we eventually test all paths? All the various env var switches, without providing an URL, switching between full and branch deployments, with a timeout, etc.

Which perhaps raises a potentially useful idea - it'd be nice if our `stub_command` could also set its exit code (defaulting to 0).

Reply:
Yes I do want to cover all paths eventually. Alternatively, we could reduce the surface area of these scripts, eg have only one way of providing inputs and do some validation in the script.

Good point about having the stub set an exit code.